src/Behat/MinkExtension/Context/MinkContext.php

```diff
@@ -483,4 +516,32 @@ protected function fixStepArgument($argument)
     {
         return str_replace('\\"', '"', $argument);
     }
+
+    /**
+     * Try to find value in element and retry for a given time
+     * @param Element $element
+     * @param string  $value
+     * @param integer $timeout
+     */
+    protected function findOrRetry(Element $element, $value, $timeout = 10000)
```

Comment:
Instead of this you can use `NodeElement::waitFor` method - https://github.com/minkphp/Mink/blob/master/src/Element/Element.php#L115 . It's exactly designed to wait for an element to appear before timeout is reached. In case if element is found earlier it's returned to the user.